### ISSUE TITLE
Legend state doesn't work in react strictmode

### ISSUE THREAD

**Issue by nouman2075** [LegendApp/legend-state#260]
In react strict mode, component doesn't renders although state is updated.

```javascript
import { observable } from "@legendapp/state";
import { enableReactTracking } from "@legendapp/state/config/enableReactTracking";

enableReactTracking({ auto: true });

const state$ = observable({ count: 0 });

function App() {
  // This makes this component responsive to test changing
  const count = state$.count.get();

  return (
    <button
      onClick={() => {
        state$.count.set((prev) => prev + 1);
        console.log(state$.count.peek());
      }}
    >
      Increment {count}
    </button>
  );
}

export default App;
```

[codesandbox](https://codesandbox.io/p/devbox/legend-state-bug-qy85x6?file=%2Fpackage.json%3A25%2C45&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clucz2tm700073b6iapehy3ov%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clucz2tm700023b6ib3xu7e59%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clucz2tm700043b6ibpdkjm6t%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clucz2tm700063b6i7tg3ue4a%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clucz2tm700023b6ib3xu7e59%2522%253A%257B%2522id%2522%253A%2522clucz2tm700023b6ib3xu7e59%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clucz44t100023b6hpaievax3%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A25%252C%2522startColumn%2522%253A45%252C%2522endLineNumber%2522%253A25%252C%2522endColumn%2522%253A45%257D%255D%252C%2522filepath%2522%253A%2522%252Fpackage.json%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clucz44t100023b6hpaievax3%2522%257D%252C%2522clucz2tm700063b6i7tg3ue4a%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clucz2tm700053b6in55qx0xs%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A5173%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clucz2tm700063b6i7tg3ue4a%2522%252C%2522activeTabId%2522%253A%2522clucz2tm700053b6in55qx0xs%2522%257D%252C%2522clucz2tm700043b6ibpdkjm6t%2522%253A%257B%2522id%2522%253A%2522clucz2tm700043b6ibpdkjm6t%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522id%2522%253A%2522clucz86uf00fs3b6h4n21mrt0%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clucz86uf00fs3b6h4n21mrt0%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

**Comment by jmeistrich** [comment]
Wow that is wild. I tried running it on older versions and tried using `observer` or `useSelector` instead of `enableReactTracking`. And nothing works. But if I trigger a fast refresh then it starts working...

If I copy the code into my React or Next.js projects it works fine?

Is this happening for you in regular projects or is it just in codesandbox? I wonder if codesandbox is doing something different that's making it not work? I know it works in Sandboxes but I've never tried making a Devbox. Maybe it's doing something different?

**Comment by nouman2075** [comment]
I tested it on my machine before i creating a sandbox. Yeah it only works on fast refresh.

**Comment by jmeistrich** [comment]
Hmm, is this something specific to vite? Is anyone else encountering this? I copied your code into a React project and a Next.js project and can't reproduce it.

I am reproducing it if I download the project and run it locally with vite, but I have no experience with vite so I'm having trouble figuring out why it's happening. When I try to step into the observable code it seems the sourcemaps are broken, and I'm seeing this error in the terminal:

`Sourcemap for "/Users/jay/Downloads/Legend State Bug/node_modules/.pnpm/@legendapp+state@2.1.10_react@18.2.0/node_modules/@legendapp/state/react.mjs" points to missing source files`

But I don't know why that would be happening. If you (or anyone else) can help me get the sourcemaps working I can dive in.

I think you added 10 to the version numbers 😅. There were some changes between 2.1.8 and 2.1.10 (the latest) that could potentially have broken it, but it's hard to investigate without being able to step through code.

**Comment by nouman2075** [comment]
Sorry that's typo in writing version. It's `2.1.9` and `2.1.10`. I also tested it in 2.1.8 and it's still not working. 
I do not have any idea about vite sourcemaps. For now, i have to disable react strict to continue my work. 
But this issue should be solved as it affects the development experience

**Comment by pongstr** [comment]
i just had the same issue when upgrading to `>2.1.9` with `vite 5.x`, doesn't work on dev mode but everything seems fine on preview/production mode.

**Comment by jmeistrich** [comment]
I finally reproduced it! Working on a fix.

**Comment by jmeistrich** [comment]
Fixed in 2.1.11

Thanks for the help tracking it down, and please let me know if you encounter this again!
